cloud/openstack/os_subnet.py

```diff
@@ -271,7 +278,14 @@ def main():
 
     try:
         cloud = shade.openstack_cloud(**module.params)
-        subnet = cloud.get_subnet(subnet_name)
+        if project is not None:
+            project_id = cloud.get_project(project)['id']
```

Comment:
This throws an exception if the project is invalid.

`TypeError: 'NoneType' object has no attribute '__getitem__'`